### DISCUSSION TITLE
Snap-7 and its effect on PLC cycle time

### DISCUSSION THREAD

**Discussion by SachinYadav74** [gijzelaerr/python-snap7#456] [Q&A] [answered]
when PLC is executing it's program and suppose it has cycle time of 30mS. When we try to get data from PLC using Snap-7, will that have effect on PLC cycle time.? Will it increase the cycle time from 30mS to 40mS?

**Comment by spreeker** [comment]
Hai,

I think the networking is done separately of the cpu in its own 'module'.

I wrote a program te read basically 'everything' from a cpu taking up to a
second. No problems.

But then again the PLC's I worked with never had a super busy complex
program running. Just a lot of stuff connected.

I would just test it out! And there is probably some documentation out
there describing the behaviour of your plc version and setup.

Stephan.

**Comment by gijzelaerr** [comment, marked as answer]
Closing as answered. PLC networking is typically handled by a separate communication module and should not significantly impact cycle time.
